### PR TITLE
Implement Toast Notification

### LINKED ISSUE #55: 15. As a user, I want to receive alerts for important actions and notifications in a non-intrusive manner
### User Story

As a user, I want to receive alerts for important actions and notifications in a non-intrusive manner, so that I can stay informed without disrupting my workflow.

### Acceptance Criteria

- **Notification Types:** The application should support different types of notifications (e.g., success, error, info, warning).
- **Positioning:** Notifications should appear at the top-right corner of the screen (or as specified) to ensure they are easily noticeable without obstructing content.
- **Duration:** Notifications should automatically disappear after a set duration (e.g., 3-5 seconds) unless the user dismisses them.
- **Custom Messages:** The user should be able to see customizable messages in the toast notifications (e.g., "Item added successfully!", "Error: Unable to delete the item.").
- **Accessibility:** The toast notifications should be accessible, allowing screen readers to read the messages.
- **Dismiss Action:** Users should have the option to manually dismiss notifications by clicking an “X” button on the toast.
- **Animation:** Notifications should fade in and out smoothly to enhance user experience.

### Implementation Details

- Use the **React Toastify** library (or a similar library) for creating and managing toast notifications.
- Integrate the toast notifications into key actions within the application, such as:
  - Successfully adding or updating items.
  - Errors occurring during API calls.
  - Important informational messages, such as reminders or tips.
- Ensure that the toast notifications are integrated in a way that they can be easily triggered from various components of the application.

### PR DESCRIPTION
**Pull request by warrenchan13**
_For an example of how to fill this template out, [see this Pull Request](https://github.com/the-collab-lab/tcl-3-smart-shopping-list/pull/44)._

## Description
This code provides alerts for important actions and notifications in a non-intrusive manner using toastify.

## Related Issue

Closes #55 

## Acceptance Criteria

- [x] Notification Types: The application should support different types of notifications (e.g., success, error, info, warning).
- [x] Positioning: Notifications should appear at the top-right corner of the screen (or as specified) to ensure they are easily noticeable without obstructing content.
- [x] Duration: Notifications should automatically disappear after a set duration (e.g., 3-5 seconds) unless the user dismisses them.
- [x] Custom Messages: The user should be able to see customizable messages in the toast notifications (e.g., "Item added successfully!", "Error: Unable to delete the item.").
- [x] Accessibility: The toast notifications should be accessible, allowing screen readers to read the messages.
- [x] Dismiss Action: Users should have the option to manually dismiss notifications by clicking an “X” button on the toast.
- [x]  Animation: Notifications should fade in and out smoothly to enhance user experience.

## Type of Changes
`enhancement` 

### Before
Screenshot 2024-10-01 at 1 49 01 PM
Screenshot 2024-10-01 at 1 49 20 PM

### After
Screenshot 2024-10-01 at 1 49 51 PM
Screenshot 2024-10-01 at 1 50 19 PM
Screenshot 2024-10-01 at 1 50 43 PM

## Testing Steps / QA Criteria
1. Ran NPM and signed in
2. Make sure Toastify works on key actions

### CODE REVIEW

**Comment by warrenchan13** [comment]
Since we replaced the "alerts" with Toastify, is there a way to set up tests for Toastify?

**Comment by kweeuhree** [comment]
Looks really cool, I love this change! Makes such a big difference for the app!
I see that tests need work, but other than that I approve.

**Review by kweeuhree — APPROVED** [review]

File: src/components/ListItem.jsx [hunk 1]
```diff
@@ -39,19 +41,18 @@ export function ListItem({ item, listPath }) {
...
 		}
 	};
 
 	const handleDeleteItem = async () => {
-		if (confirm(`Are you sure you want to delete this item?`)) {
-			try {
-				await deleteItem(listPath, id);
-			} catch (error) {
-				alert('Item was not deleted');
-			}
```
Comment (kweeuhree) [inline src/components/ListItem.jsx:53]: need to add this if block back to meet previously required acceptance criteria

Comment (warrenchan13) [reply src/components/ListItem.jsx:53]: done

**Comment by kweeuhree** [comment]
I noticed that you added back the if block and voted for the custom dialog! Do you think you'd want to create that dialog? Or we can team up, let me know!
Also, it looks like tests do not pass. Have you tried adding that 'spy' that the error message is talking about? I will take a look at updating the tests in a bit

File: src/App.jsx [hunk 2]
```diff
@@ -42,6 +43,16 @@ export function App() {
 
 	return (
 		<Router>
+			<ToastContainer
+				position="top-right"
```
Comment (deeheber) [inline src/App.jsx:47]: Group PR review: consider center-top

File: src/components/AddItems.jsx [hunk 3]
```diff
@@ -3,6 +3,8 @@ import { useStateWithStorage, normalizeItemName } from '../utils';
 import { addItem } from '../api';
 import TextInputElement from './TextInputElement';
 import RadioInputElement from './RadioInputElement';
+import { toast } from 'react-toastify';
+import 'react-toastify/dist/ReactToastify.css';
```
Comment (deeheber) [inline src/components/AddItems.jsx:7]: Group PR review: consider importing at the app level.

File: src/views/Home.jsx [hunk 4]
```diff
@@ -16,17 +18,20 @@ export function Home({ data, setListPath, userId, userEmail }) {
 
 		try {
 			if (currentLists.includes(listName.toLowerCase())) {
-				alert('The list already exists. Please enter a different name.');
+				toast.error(
+					'The list already exists. Please enter a different name.',
+					{},
```
Comment (deeheber) [inline src/views/Home.jsx:23]: Nit: double check if this is necessary. It might be but we're not sure.

**Review by deeheber — COMMENTED** [review]
We discussed in our sync that it's ok for @warrenchan13 to remove the test portion of this to move forward.

The other option is to add mocks for this library for tests to pass.

**Review by granolagabrielle — APPROVED** [review]

**Review by jendevelops — COMMENTED** [review]
As @deeheber said, just need to update the tests to get this baby merged in!

File: tests/List.test.jsx [hunk 5]
```diff
@@ -3,24 +3,6 @@ import { MemoryRouter } from 'react-router-dom';
 import { List } from '../src/views/List';
 import { mockShoppingListData } from '../src/mocks/__fixtures__/shoppingListData';
 
-beforeEach(() => {
```
Comment (jendevelops) [inline tests/List.test.jsx:6]: Is there a reason why you removed this whole `beforeEach`?

You should only have to remove line 21, and the test should pass.

File: tests/ManageList.test.jsx [hunk 6]
```diff
@@ -2,24 +2,6 @@ import { render, screen } from '@testing-library/react';
...
-			}),
-			setItem: vi.fn(),
-			clear: vi.fn(),
-		},
-		writable: true,
-	});
-
-	vi.spyOn(window, 'alert').mockImplementation(() => {});
-});
-
```
Comment (jendevelops) [inline tests/ManageList.test.jsx:22]: Same here,  you only have toe delete line 20. You're removing the mocked local storage value to be able to simulate the groceries list, which is why all the tests are failing

**Review by jendevelops — APPROVED** [review]
Thanks for fixing the tests!

**Pull request merged**